examples/api/lib/widgets/linked_text/linked_text.2.dart

```diff
@@ -0,0 +1,123 @@
+
+class LinkedTextApp extends StatelessWidget {
+  const LinkedTextApp({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: MyHomePage(title: 'Flutter Link Twitter Handle Demo'),
+    );
+  }
+}
+
+class MyHomePage extends StatelessWidget {
+  MyHomePage({
+    super.key,
+    required this.title
+  });
+
+  final String title;
+  //static const String _text = '@FlutterDev is our account on Twitter, or go to flutter.dev';
+  static const String _text = '@FlutterDev is our Twitter account, or find us at www.flutter.dev';
+
+  void _handleTapTwitterHandle(BuildContext context, String linkText) {
+    final String handleWithoutAt = linkText.substring(1);
+    final String twitterUriString = 'https://www.twitter.com/$handleWithoutAt';
+    final Uri? uri = Uri.tryParse(twitterUriString);
+    if (uri == null) {
+      throw Exception('Failed to parse $twitterUriString.');
+    }
+    _showDialog(context, uri);
+  }
+
+  void _handleTapUrl(BuildContext context, String urlText) {
+    final Uri? uri = Uri.tryParse(urlText);
+    if (uri == null) {
+      throw Exception('Failed to parse $urlText.');
+    }
+    _showDialog(context, uri);
+  }
+
+  void _showDialog(BuildContext context, Uri uri) {
+    // A package like url_launcher would be useful for actually opening the URL
+    // here instead of just showing a dialog.
+    Navigator.of(context).push(
+      DialogRoute<void>(
+        context: context,
+        builder: (BuildContext context) => AlertDialog(title: Text('You tapped: $uri')),
+      ),
+    );
+  }
+
+  final RegExp _twitterHandleRegExp = RegExp(r'@[a-zA-Z0-9]{4,15}');
+  final RegExp _urlRegExp = RegExp(r'(?<!@[a-zA-Z0-9-]*)(?<![\/\.a-zA-Z0-9-])((https?:\/\/)?(([a-zA-Z0-9-]*\.)*[a-zA-Z0-9-]+(\.[a-zA-Z]+)+))(?::\d{1,5})?(?:\/[^\s]*)?(?:\?[^\s#]*)?(?:#[^\s]*)?(?![a-zA-Z0-9-]*@)');
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(title),
+      ),
+      body: Center(
+        child: Builder(
+          builder: (BuildContext context) {
+            return SelectionArea(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: <Widget>[
+                  LinkedText.textLinkers(
+                    text: _text,
+                    textLinkers: <TextLinker>[
+                      TextLinker(
+                        textRangesFinder: TextLinker.textRangesFinderFromRegExp(_urlRegExp),
+                        linkBuilder: InlineLinkedText.getDefaultLinkBuilder((String urlString) {
+                          return _handleTapUrl(context, urlString);
+                        }),
+                      ),
+                      TextLinker(
+                        textRangesFinder: TextLinker.textRangesFinderFromRegExp(_twitterHandleRegExp),
```

Comment:
I wonder if we should just have `LinkedText` and `LinkedText.textLinkers` since the latter can also do `regExp` like shown in this example.

Reply:
Another option is to just have `LinkedText` with a nullable `textLinkers` property. If nothing is passed then we can use our defaults, if not then we use the given `textLinkers`. It looks like `TextLinker.defaultTextRangesFinder` is exposed so if someone wanted to use the default url regex they could.

Reply:
I'm going to leave a comment below about these constructors, I'll link to it.

Reply:
Comment with discussion about constructors: https://github.com/flutter/flutter/pull/125927#issuecomment-1699575816